lib/libc/sys/select.c

```diff
@@ -0,0 +1,56 @@
+#include <sys/select.h>
+#include <sys/time.h>
+#include <sys/event.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+
+int select(int nfds, fd_set *restrict readfds, fd_set *restrict writefds,
+           fd_set *restrict exceptfds, struct timeval *restrict timeout) {
+    int kq;
+    int ret;
+    struct timespec timeout_ts;
+    struct kevent *events;
+    int nevents = 0;
+
+    if (timeout != NULL)
+        tv2ts(timeout, &timeout_ts);
+
+    kq = kqueue();
+    if (kq < 0)
+        return -1;
+
+    events = malloc(2 * nfds * sizeof(struct kevent));
```

Comment:
Handle situation when malloc return NULL.
You have to check if nfds is a correct argument ([EINVAL](https://man7.org/linux/man-pages/man2/select.2.html))